mock/src/main.rs

```diff
@@ -0,0 +1,137 @@
+use std::{fs::File, io::Write, path::PathBuf, time::Instant};
+
+use clap::Parser;
+use crypto::{hash::rpo::RpoDigest as Digest, FieldElement, Word};
+use miden_objects::{
+    builder::DEFAULT_ACCOUNT_CODE,
+    mock::{Immutable, MockChain, OnChain},
+};
+use rand::SeedableRng;
+use rand_pcg::Pcg64;
+
+#[derive(Parser, Debug)]
+struct Args {
+    /// Value used to seed the PRNG.
+    #[arg(long)]
+    seed: Option<u64>,
+
+    /// Output file
+    output: PathBuf,
+}
+
+fn seed_to_word(seed: &str) -> Word {
+    let seed_bytes = hex::decode(seed).unwrap();
+    let data = unsafe { FieldElement::bytes_as_elements(&seed_bytes) }.unwrap();
+    let seed: Word = [data[0], data[1], data[2], data[3]];
+
+    seed
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let args = Args::parse();
+
+    let small_rng = match args.seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+
+    println!(
+        "Generating mock data, results will be saved to {}",
+        args.output.to_str().expect("invalid file name")
+    );
+
+    let mut output = File::create(args.output)?;
+
+    // Block 1
+    // --------------------------------------------------------------------------------------------
+    let mut mock_chain = MockChain::new(small_rng);
+    let start = Instant::now();
+
+    let faucet = mock_chain.build_fungible_faucet_with_seed(
+        seed_to_word("3ef982ea3dca3f89179e1e86ef1c263896ef1e07e76325ce7c69825046bf75ec"),
+        OnChain::Yes,
+        DEFAULT_ACCOUNT_CODE,
+        Digest::default(),
+    );
+    println!("Fungible faucet created {} [took: {}s]", faucet, start.elapsed().as_secs());
+    mock_chain.seal_block();
+
+    // Block 2
+    // --------------------------------------------------------------------------------------------
+    let amount = 100;
+    let asset = mock_chain.build_fungible_asset(0, amount);
+
+    let storage = vec![];
+    let assets = vec![asset];
+    let start = Instant::now();
+    let account0 = mock_chain.build_account_with_seed(
+        seed_to_word("fbee7d33ae8550dfde6d63ad899177d42382f8a7f04b55a5a57ee6a30a2c2a52"),
+        DEFAULT_ACCOUNT_CODE,
+        storage,
+        assets,
+        Immutable::No,
+        OnChain::No,
+    );
```

Comment:
For the kernel not to interpret this as a "new" account, we need to make sure that the nonce is not set to `0` (e.g., could be set to `1`). Not sure if this is currently happening.

Also, @frisitano - do we already have this check implemented?